sdk/include/opentelemetry/sdk/trace/tracer.h

```diff
@@ -46,6 +66,7 @@ class Tracer final : public trace_api::Tracer, public std::enable_shared_from_th
 
 private:
   opentelemetry::sdk::AtomicSharedPtr<SpanProcessor> processor_;
+  opentelemetry::sdk::AtomicSharedPtr<Sampler> sampler_;
```

Comment:
Does the sampler have to be a shared instance? Or can each Trace object own a separate instance of the sampler?

Reply:
The sampler has to be shared and global. This will enable the SDK to do proper probability sampling across tracers.

Reply:
When we only set this on initialization and remove all setters for the sampler, we could make this a `const std::shared_ptr<Sampler>`.